### DISCUSSION TITLE
Road to the Next Update! - Future Development and PR Merging Strategy

### DISCUSSION THREAD

**Discussion by gozarman** [Gozargah/Marzban#985] [Announcements]
### Dear Marzban Community,

Firstly, we want to extend our heartfelt gratitude to each and every one of you for your continued support, contributions, and active participation in our project. It is truly an honor to have such a vibrant and dedicated community that shares our passion and vision. We want to inform you about our thoughts and plans for the future of the project, and let you know that we have read your comments and feedback.

## Current Focus: UI and Feature Enhancements
Our primary focus at the moment is on polishing up the user interface (UI) and making Marzban more extendable. This involves a comprehensive redesign and enhancement of the current system to ensure it is capable of supporting a wide range of features. Our goal is to create a platform that is not only powerful but also extendable and understandable for users and developers from all around the world.

We are excited to announce that our latest changes to the UI are available in the [next](https://github.com/gozargah/marzban/tree/next) branch. We encourage you to check out these updates and provide your feedback. This branch includes the initial steps towards a more polished and feature-rich interface.

## Why Some PRs Are Not Merged
We greatly appreciate all the contributions made by our community. However, it is essential for us to align all changes with our long-term vision and strategy. Here are a few key points to consider:

- **Consistency and Compatibility**: We aim to maintain a consistent codebase that ensures compatibility with future updates. Some PRs, while valuable, may not fit into our current roadmap or may introduce changes that could complicate future development.
- **Quality and Extensibility**: We strive to ensure that all new features and changes are of high quality and can be easily extended. This often requires us to carefully evaluate and sometimes rework proposed changes to fit our architecture and design principles.
- **Long-Term Vision**: We have a clear vision for the future of Marzban, which includes specific features and improvements. All contributions need to align with this vision to ensure that we are moving towards our goals efficiently. We will discuss more about our vision in future discussions.

## Temporary Hold on PR Merges
Currently, we are focused on polishing the UI, which involves significant changes not only to the frontend but also to the backend. This makes it challenging to merge PRs that affect either the backend or frontend until these modifications are complete. We will check and merge PRs as soon as the UI enhancements are finalized.

We are committed to migrating our current UI to the new UI as soon as possible. Once this migration is complete, we will resume maintaining and merging PRs. Please be patient and wait until the UI is ready.

## Our Commitment and Your Feedback
We are committed to working on the Marzban long-term and maintaining it to the best of our capabilities. Here’s what you can expect in the near future:

- **UI Enhancements**: A more intuitive and visually appealing user interface that enhances user experience.
- **Feature Expansion**: New features that will be added to improve the quality of the users experience.
- **Better Documentation**: Comprehensive documentation to help users and contributors understand the system better and contribute more effectively.

We value your feedback and encourage you to continue sharing your ideas and concerns. Your input is crucial in shaping the future of Marzban. However, we need your thoughts and comments to be clear and straightforward. Please follow the templates when you are opening an issue or discussion to help us manage them more effectively.

## Conclusion
Thank you once again for your understanding, support, and active participation. Together, we can make Marzban even better and more impactful. Please feel free to join the discussion below and share your thoughts.

Best regards,

Marzban core team (Saint, Gozarman).

**Comment by erfjab** [comment]
Thank you very much. 
Can we give you an idea for the UI with the concepts?

**Reply by gozarman** [reply]
Hey,
Sure, We are so pleased to know your idea! 
Just please check the `next` branch first to find a general idea about our current changes if possible.

**Reply by erfjab** [reply]
![image](https://github.com/Gozargah/Marzban/assets/143827987/3e8d2663-999d-459b-91a5-c2aebf5214a1)

a few weeks ago, I came up with this concept in just ten minutes. The logic I had in mind was to have the statistics section provide us with data on system metrics, users, nodes, and admins. The users section is for creating users, admins section for creating admins, and the stink section for handling hosts, nodes, and cores. This way, page layout could make development more efficient and easier.

**Reply by mikeesierrah** [reply]
<img width="331" alt="Screenshot 2024-05-16 214642" src="https://github.com/Gozargah/Marzban/assets/144580664/5eba3ce0-86f8-4969-87c4-555cae825f1c">

THIS IS BETTER

**Reply by gozarman** [reply]
Looks fantastic, this is really close to what we were planning about the statistics page. Here's a not-finalized version of statistics page. Font sizes, the chart, and the cards are not finalized. But only to show you a concept of what we expect about this page:

![Statistics](https://github.com/Gozargah/Marzban/assets/120019621/3cda77ea-d74b-4012-a38e-33a94fc4e531)

As you can see in the picture, there is a dropdown on top of the page which makes it possible for admin to see the statistics of a single node.
Let us know what you think about this design and what more needs to cover in this page.
We will be happy if you provide a list of the stats that would be useful if we show them in this page.

**Reply by xmohammad1** [reply]
what do you think about transfer all data usages to a separate page ? , like we can have admins data usage list + daily usages and other info about data usages there
especially data usage in iran matters cuase most of Iranian datacenters take payment per GB :))

**Reply by mahanhpn** [reply]
Wow, it's just amazing .

**Reply by xmohammad1** [reply]
i made some type of conncept for admin usage info
1.
![image](https://github.com/Gozargah/Marzban/assets/117955130/2111773d-3216-4bfd-98d7-6e3ecb63514f)
2.
![image](https://github.com/Gozargah/Marzban/assets/117955130/24cb7ed8-a5ef-4d9c-b567-038b38dffda7)
3.
![image](https://github.com/Gozargah/Marzban/assets/117955130/54faf006-bd0a-487d-9fa3-8edd005c1f34)

**Reply by miladia13** [reply]
wow! Fantastic :)

**Comment by dailymobile** [comment]
مرسی از زحماتت

**Comment by mikeesierrah** [comment]
Finally answers , we appreciate and acknowledge your hardwork , thank you.

Is there an estimated time for next branch finalization?
Not complete release but at-least beta release

**Reply by gozarman** [reply]
Unfortunately there is no accurate estimated time for the beta release of the next branch but we are doing our best to make it ready in 1 month from now.

**Comment by karimzzzm** [comment]
کارت عالیه یعنی کاری کردی همه کسای که بابت پنل پولی میدادن اومدن سمت مرزبان دارن استفاده می کنن راضی ترم هستن 
فقط اگر تونستی یه نگاهی به xmplus بنداز ازش ایده بگیر 
من استفده نمی کنم از مرزبان ولی دمت خیلی گرم واقعا

**Comment by DigneZzZ** [comment]
Hello!
I'm eagerly anticipating the updated UI. The suggestions are not entirely novel but may be accepted for review by the team (or some of its members).
Proposed enhancements for the new interface:

- Implementing bulk actions on users.
- Exploring an alternative approach to grouping inbounds, avoiding automatic addition to users when a new inbound is created.
- Incorporating maximum API options into the interface.
- Making the statistics page clickable.
- Utilizing user templates during user creation/editing (similar to a Telegram bot).
- Fine-tuning user templates (as mentioned earlier).
- Defaulting to installation with MySQL.
- Enabling node creation from the main server interface.
- Are there any plans to explore setting up x-ray from the interface (without coding)?

**Reply by gozarman** [reply]
Great features, could you please open a "[Feature request](https://github.com/Gozargah/Marzban/issues/new?assignees=&labels=enhancement&projects=&template=feature_request.md&title=Feature+title)" issue and mention these features there? I'm afraid we miss them here.

**Reply by DigneZzZ** [reply]
done
https://github.com/Gozargah/Marzban/issues/986

**Comment by raminsajedi** [comment]
Thank you and all other members of Marzban project who are developing such a great tool for free so that in these difficult days, it can be helpful for us and all the people of Iran to access the free internet.
In hope of freedom.
best regards

**Comment by xmohammad1** [comment]
hi there , hope u guys doing well :)
i have some suggestion 
1. enter , client in how many days expire , instead of date
2. bring policy to xray core #975 
3. Ability to change customer usernames
4. auto refresh in users list
5. hide settings for nonsudo admins

**Reply by gozarman** [reply]
Hi there, 
Thanks for your suggestions.

I just didn't get the point of the first suggestion. Specifying a date isn't more general(which can cover more use cases) instead of specifying the expiry date based on how many days?

**Reply by xmohammad1** [reply]
well , my main idea was creating on hold clients , like we add 30 days and exipre time start after first use

**Comment by abedin70** [comment]
ممنون از شما بابت همه زحماتی ک کشیدید.
فوق العاده هستید همیشه

**Comment by idaniali** [comment]
خسته نباشید بابت زحماتی که میکشید .
امکان اضافه کردنه گزینه شروع زمان از اولین اتصال هست ؟
و به جای اینکه تاریخ رو باز کنه ، فقط بزنی ۳۰ روز . اینطوری برا محاسبه از اولین اتصال هم راحت تره

**Comment by ImMohammad20000** [comment]
سلام
خسته نباشی
 بنظرت بهتر نیست که حداقل با یه آپدیت با آخرین تغییرات xray هماهنگ بشیم تا بخاد فرانت جدید آماده بشه؟؟
بنظرم اینجوری خیلی بهتره چون ظاهرا تا فرانت جدید آماده بشه راه زیادی داریم

**Reply by gozarman** [reply]
Hey dude,
We are trying our best to make the beta version of the next branch ready for the next month.

**Reply by ImMohammad20000** [reply]
Thanks for answering but 1 month is still too much it's bother to have a version 0.4.10 with latest changes of xray

We already have pull requests for latest changes only merge them🙏

These backend stuff required no changes to current and next frontend

**Reply by ImMohammad20000** [reply]
And still waiting...😶

**Comment by rezazoom** [comment]
Thank you for the update and for sharing the exciting progress on the new UI enhancements. I'm looking forward to exploring the next branch and providing feedback.
I have a couple of questions regarding the upcoming updates:
- Mobile-Friendliness: Will the new UI changes include improvements to make Marzban more mobile-friendly compared to the current version?
- Update Checker and Auto-Updating Feature: Can we expect the next update to include a built-in update checker and auto-updating feature for seamless maintenance?

**Reply by gozarman** [reply]
Hi there,
Thanks for your feedback.

Mobile friendliness: All of the pages are responsive. So the pages should look fine in small devices (mobiles).
Auto update: It is not planned for the beta version of the next version, but it seems to be a really useful feature. Please submit this in the issues as a "[Feature request](https://github.com/Gozargah/Marzban/issues/new?assignees=&labels=enhancement&projects=&template=feature_request.md&title=Feature+title)".

**Reply by rezazoom** [reply]
Thanks, wrote in #1001

**Comment by soheylshorg** [comment]
خیلی مچکرم از گذر من گرامی بابت انجام و وقتی که برای این پنل که کامل هم بود گذاشتید تا دسترسی ما به بهترین شکل ممکن باشد 
باتشکر از تیم توسعه مرزبان

**Comment by sarboy1371** [comment]
مرسی از تیم قدرتمند مرزبان
آپدیت زود به زود لطفا😉

**Comment by Mititeryaki** [comment]
اگر روشی اد کنید بشه با ipv6 نود اد داد خیلی بهتر میشه

**Reply by gozarman** [reply]
Hi there

Please submit it as a "[Feature request](https://github.com/Gozargah/Marzban/issues/new?assignees=&labels=enhancement&projects=&template=feature_request.md&title=Feature+title)"  issue.
Thanks

**Comment by Non-believer** [comment]
ضمن عرض تشکر از زحمات و تلاش های شما حقیقتش اگر یک بخش به اسم queries به پنل اضافه شه تا بتونیم راحت به کوئری های کاربردی دسترسی داشته باشیم عالی میشه.

**Reply by gozarman** [reply]
What do you mean about the "queries"? What are these queries supposed to do?

**Reply by ImMohammad20000** [reply]
این بخش مربوط به دیتابیسه و ربطی به پنل نداره

**Reply by xmohammad1** [reply]
برای کوئری باید بری داخل phpmyadmin یا با sqlite3 وصل بشی به دیتابیست 
 چیزی نیست که به پنل ربط داشته باشه

**Comment by ghost** [comment]
Thank you for your hard work, I wish you would bring a feature that backup is not limited to the Telegram bot

**Comment by Midaga11** [comment]
درود لطفا پروتکل kcp هم فعال کنین براش واقعا خیلی خوبه درخواست ها هم براش زیاده اگه گپ رو نگاه کنین

**Comment by zaskochill** [comment]
Thank you for the work you have done! Really looking forward to the new version.
Here are a few suggestions of what we would like to see:

## 1) Add the ability to select custom configs for sing-box and v2ray-json when creating and editing a user:

If we add CUSTOM_TEMPLATES_DIRECTORY, we can now only add 1 config file for singbox and for v2ray-json. 
I would like to make it possible to add multiple configs and then give the option to select a config for each user.
That is, if we give a user a subscription with sing-box or v2ray-json, we are limited to one configuration and cannot give the user a different config.
That said, we could make a variable SINGBOX_SUBSCRIPTION_DIRECTORY and V2RAY_SUBSCRIPTION_DIRECTORY, then put some config files in there, set the default configs also in the settings and let the user choose the config.
In this way we can make configs:
Standard, For China, For Iran, For Russia, With ad blocking, Without ad blocking and so on.
I have attached an example of how this can be implemented in the UI.

## 2) Add the option to select a node for each user
I think it would be nice to extend the Node feature to allow users to be given Nodes separately.
That is, now nodes take the config from the main server and duplicate all protocols.
It would be very convenient when creating or editing a user to be able to enable/disable the node. Also, you can open the parameters of each node and select the necessary protocols for each node.
At the moment the only way to give a user a separate node is to make a new inbound for it, which is not convenient.
An example of how this could be done is also attached in the screenshot below.

## 3) Combine protocols into groups.
Give the user not individual protocols, but groups.
In a separate menu it is possible to edit and add new groups.
This way, by changing the group in one place, you can quickly add or remove protocols from many users at once

## 4) Make it possible to create users with ON_HOLD status

Thank you for your hard work! We are very much looking forward to updates 

![design_marzban](https://github.com/Gozargah/Marzban/assets/47404915/a7eee926-bed0-4669-b1d6-e0d240f079e5)

**Reply by zaskochill** [reply]
## 5. Add logic to update the subscription when the subscription has “Expired”:
When a user's subscription expires, remove all protocols from the subscription and leave one “YOUR SUBSCRIPTION EXPIRED”.
And also, it would be nice to be able to change this informational message from the settings.
![IMG_0968](https://github.com/Gozargah/Marzban/assets/47404915/4c6edfc3-9f17-4c0f-85e1-90d31a8c8205)

**Comment by nsa14** [comment]
قربون همه شما زحمت کشان پنل عالی مرزبان
واقعا بینظیر هستید

چندتا مورد اگر اضافه و تغییر داده بشه عالیه:
۱-در قسمت تاریخ، یه باکسی اضافه بشه که بتونیم تاریخ رو بصورت روز هم اضافه کنیم مثلا ۳۰ روز انتخاب یا وارد کنیم
یا بحاش دکمه های +1, +3 و اینجوری گذاشته بشه ک با کلیک روش اتوماتیک بکماه و یه ماه درج بشه در باکس تاریخ

۲-در صفحه نمایش موبایل حالت ریسپانسیو برای پنجره های popup یکم ایراد داره و بعضی وقتها دکمه ذخیره تغییرات دیده نمیشه. امکان دیده شدن همه گزینه ها در پنجره نیست و باید مجدد رفرش بشه صفحه تا دیده بشه.

۳-در پنجره ساخت اکانت حذید، گزینه Periodic Usage Reset در زیر کادر انتخاب تاریخ گذاشته شود بهتر است

۴-*** نمایش تعداد کاربر متصل به هر کانفیگ -مورد خیلی مهم و کاربردیه

۵-در صفحه نمایش QR زیر عکس ها اسم کانفیگ نوشته شود که مشخص بشه برای کدومه

۶-با کلیک روی عکس QR ، کد کانفیگ کپی شود

۷-اضافه کردن پروتکل های kcp,httpupgrade

۸- تو قسمت نمایش سرتیفیکیت، یه دکمه کوچیک‌برای کپی گذاشته بشه کنار دکمه نمایشش

۹-کادر های Active Users، Data Usage و بقیه، سایزشون کوچیکتر بشه تا فضای بیشتر به نمایش کاربرا داده بشه

۱۰- تعداد نمایش ایست کاربرا ، عدد های ۵۰ و ۱۰۰ هم اضافه بشه

از زحمات شما تشکر میکنم و کاری از دیت ما هم بر بیاد درخدمت هستیم عزیزان🙏

**Comment by fx505** [comment]
سلام و خسته نباشید میگم به همه‌ی دوستان
لطفا تو ui جدید qrcode اپ‌های سینگ باکس رو هم ساپورت کنین، یه نمونه‌ش رو برای مثال میذارم چون کمی فرق داره (فقط لینک تبدیل نمیشه مثل اکثر کلاینت‌ها)
![Screenshot_20240604-002810_sing-box](https://github.com/Gozargah/Marzban/assets/158324614/1111cd2c-fefd-4391-a4d5-1ae5c596aaf6)

**Comment by xmohammad1** [comment]
@gozarman 
hey man , how are you doing? 
its been little more than one month , any news on it?

**Comment by ahwazi12** [comment]
سلام وخسته نباشی  
اگه میشه  برای  محدکد کردن ادمین ها  هم  بصورت  حجمی  یا تعداد ساخت یوزر 
براشون درنظر بگیرد
